packages/doenetml-worker-javascript/src/test/baseComponent/renderedNames.test.ts

```diff
@@ -0,0 +1,532 @@
+import { describe, expect, it, vi } from "vitest";
+import { createTestCore } from "../utils/test-core";
+
+import {
+    renderedNameSortFunction,
+    type PotentialNames,
+} from "../../utils/renderedNames";
+import { updateBooleanInputValue } from "../utils/actions";
+
+const Mock = vi.fn();
+vi.stubGlobal("postMessage", Mock);
+vi.mock("hyperformula");
+
+describe("Rendered name tests", async () => {
+    it("find all potential names", async () => {
```

Comment:
This test and the next one are for internal implementation details. They should probably be moved to their own test file and labeled as testing that exact function. There should be a comment that they can be deleted if permalinks are created a different way.